File: app/goals/goal.py
from datetime import datetime


class Goal:
    def __init__(
        self,
        goal_id,
        name,
        target_amount,
        current_amount=0.0,
        target_date=None,
        created_at=None
    ):
        self.id = goal_id
        self.name = name
        self.target_amount = target_amount
        self.current_amount = current_amount
        self.target_date = target_date
        self.created_at = created_at or datetime.now()

File: frontend/frontend/src/types/goal.ts
// NOT backed by Python yet. Forward-looking contract based on the product spec.

export interface FinancialGoal {
  id: number;
  name: string;
  target_amount: number;
  current_amount: number;
  target_date: string; // ISO date
}

export interface FinancialGoalWithProgress extends FinancialGoal {
  remaining: number;
  percentComplete: number;
}

export type NewFinancialGoal = Omit<FinancialGoal, 'id' | 'current_amount'> & {
  current_amount?: number;
};
export type UpdateFinancialGoal = Partial<Omit<FinancialGoal, 'id'>>;
